lib/vehicle.rb

```diff
@@ -22,4 +26,12 @@ def antique?
   def electric_vehicle?
     @engine == :ev
   end
+
+  def change_registration_date 
+    @registration_date = Date.today
+  end
+
+  def change_plate_type(plate_type)
```

Comment:
Yes, I like this! Much better than an attr_accessor since now the class itself is in charge of making those attribute changes.